code/modules/reagents/Chemistry-Recipes/Others.dm

```diff
@@ -357,4 +404,18 @@
 	id = "saltpetre"
 	result = "saltpetre"
 	required_reagents = list("potassium" = 1, "nitrogen" = 1, "oxygen" = 3)
-	result_amount = 3
\ No newline at end of file
+	result_amount = 3
+
+/datum/chemical_reaction/lye
+	name = "lye"
+	id = "lye"
+	result = "lye"
+	required_reagents = list("sodium" = 1, "hydrogen" = 1, "oxygen" = 1)
+	result_amount = 3
+
+/datum/chemical_reaction/lye2
+	name = "lye"
+	id = "lye"
+	result = "lye"
+	required_reagents = list("ash" = 1, "water" = 1)
+	result_amount = 2
```

Comment:
Since lye+water=space cleaner, you'll never be able to produce lye this way, unless you do it one unit at a time.